tests/integration/ha/helpers.py

```diff
@@ -172,3 +173,37 @@ async def assert_continuous_writes_consistency(c_writes: ContinuousWrites) -> No
 
     assert result.max_stored_id == result.count - 1
     assert result.max_stored_id == result.last_expected_id
+
+
+def find_process_by_port(port: int = 9200) -> Optional[int]:
+    """Find pid of process listening on port."""
+    for proc in psutil.process_iter():
+        for conn in proc.connections():
+            if conn.status == "LISTEN" and conn.laddr.port == port:
+                return proc.pid
+
+    return None
+
+
+async def send_kill_signal_to_process(
+    ops_test: OpsTest, app: str, unit_id: int, signal: str
+) -> None:
+    """Run kill with signal in specific unit."""
+    # Killing the only instance can be disastrous.
+    if len(ops_test.model.applications[app].units) < 2:
+        await ops_test.model.applications[app].add_unit(count=1)
+        await ops_test.model.wait_for_idle(
+            apps=[app],
+            status="active",
+            timeout=1000,
+            idle_period=IDLE_PERIOD,
+        )
```

Comment:
I would prefer to have this logic in the test itself, not hidden in kill function